src/components/SearchViaISBN.js

```diff
@@ -19,22 +18,19 @@ export default function SearchViaISBN({ onHandleSetSearchedBook }) {
       }, 5000)
       return () => clearTimeout(timer)
     }
-  }, [errorMessage, bookISBN, history])
+  }, [errorMessage, history])
 
-  function handleSubmit(event) {
+  async function handleSubmit(event) {
     event.preventDefault()
     const form = event.target
     const { isbn } = form.elements
-    getBook(isbn.value.replace('-', ''))
-      .then(onHandleSetSearchedBook)
-      .then(() => {
-        setBookISBN(isbn.value)
-        if (!bookISBN) {
-          setErrorMessage('ISBN error')
-        } else {
-          history.push('/add-book-form')
-        }
-      })
+    try {
+      const book = await getBook(isbn.value.replace('-', ''))
+      onHandleSetSearchedBook(book)
+      history.push('/add-book-form')
+    } catch (error) {
+      setErrorMessage('ISBN error')
+    }
```

Comment:
got to love async; iMo more readable and way easier to write

Reply:
I don't know all the other options the coding world has to offer, but I also like async. Even so it took me longer to get to this solution. I have to practice async await more. 😅